### PR TITLE
`Exiled.Launcher`

### PR DESCRIPTION
**Pull request by Jesus-QC**
Here we go.

`Exiled.Updater`, `Exiled.Patcher`, `Exiled.Bootstrap` and `Exiled.Installer` will be deprecated by this pull request, merging all into one.

The Launcher will:
- Install exiled with version provided.
- Inject exiled bootstrap.
- Run LocalAdmin

### Need to add version checker to bootstrap

# Almost Finished
image

### CODE REVIEW

**Comment by Monaldcry7788** [comment]
Does it will support multiadmin?

**Comment by Jesus-QC** [comment]
> Does it will support multiadmin?

Yeah, ofc, but u will have to write smt like `--starting-point MultiAdmin.exe`

File: Exiled.Launcher/Features/Updater/InstallManager.cs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        Console.WriteLine("Fetching Releases:");
+        try
+        {
+            using HttpResponseMessage res
+                = httpClient.GetAsync($"https://api.github.com/repos/{RepoID}/releases?per_page=100").GetAwaiter().GetResult();
+
+            if (res.StatusCode == HttpStatusCode.TooManyRequests)
+            {
+                Console.WriteLine("You have been ratelimited by github, therefore Exiled couldn't be installed.");
+                Console.WriteLine("Skyping installation.");
```
Comment (NaoUnderscore) [inline Exiled.Launcher/Features/Updater/InstallManager.cs:124]: SKYPING LMAO

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

File: Exiled.Launcher/Features/Updater/InstallManager.cs [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        {
+            Console.WriteLine("Couldn't find the file Exiled.tar.gz in the release.");
+            return false;
+        }
+
+        try
+        {
+            Console.WriteLine("Downloading the release.");
+            string tmpDownload = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), ExiledAssetName);
+            Console.WriteLine("Dest url: " + asset.BrowserDownloadUrl);
```
Comment (BuildBoy12) [inline Exiled.Launcher/Features/Updater/InstallManager.cs:79]: Dest

File: Exiled.Launcher/Features/Updater/InstallManager.cs [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+            using HttpResponseMessage res = httpClient.GetAsync(asset.BrowserDownloadUrl).GetAwaiter().GetResult();
+
+            if (res.StatusCode == HttpStatusCode.TooManyRequests)
+            {
+                Console.WriteLine("You have been rate limited by github, therefore Exiled couldn't be installed.");
+                Console.WriteLine("Skyping installation.");
```
Comment (BuildBoy12) [inline Exiled.Launcher/Features/Updater/InstallManager.cs:85]: Skyping

**Comment by Jesus-QC** [comment]
LocalAdmin is receiving an update to handle plugins, we shouldn't have anything that overrides localadmin.

**Comment by Jesus-QC** [comment]
Oh wait, forgot this launched localadmin after lol

**Comment by IRacle1** [comment]
🙃

**Pull request closed without merging**